### DISCUSSION TITLE
AUDIO FILTER REGISTERS

### DISCUSSION THREAD

**Discussion by WHIZZKIIDD** [egzumer/uv-k5-firmware-custom#599] [General]
Hello!  I have tried to find settings to disable or change the low-pass options for RX, Is there any way to change the lowpass audio filter to 4.5kHz ??  And TX too if possible. I would like to extend to TX-RX 4.5KHz for a repeater. When fpassing through audio using the same Fc, The -3dB point becomes -6dB.   I want to configure it to maximum audio frequency range to fully preserve the original Audio bandwidth

**Comment by ValdikSS** [comment]
Take a look at this mod: https://github.com/mobilinkd/uv-k6-digital-mod/blob/master/UV-K6.ipynb

Overview of Changes

The hardware changes which are documented below allow passing low-frequency, wide, flat baseband between the digital modem and the radio's RF chip. This is an absolute necessity for digital baseband modulation.

> The firmware changes adds a new Digital (DIG) mode which does the following:
> 
>     Eliminate pre-emphasis on transmit and de-emphasis on receive.
>     Eliminate the TX & RX 300Hz sub-audio channel high-pass filter which is used for CTCSS and DCS.
>     Eliminate the TX & RX 3000Hz low-pass audio filter.
>     Bypass the transmit low-pass filter.
>     Disable the MIC AGC and the ALC (limiter).
>     Disable squelch.
>     Keep MIC ADC enabled in RX mode to establish and maintain proper DC bias of the MIC input.
>     Enable MIC ADC during init for a short period to establish proper DC bias of the MIC input.
>     Reduce turn-around time from ~140ms to less than 80ms by leaving the audio amplifier enabled during TX. This also reduces a DC offset error at start of RX. The cost is some wasted current use during TX.
> 
> Note that the EGZUMER firmware, on which these changes are based, already reduces the turn-around time of the stock UV-K6 firmware from over 350ms to around 140ms.

**Reply by WHIZZKIIDD** [reply]
thanks man ... actually i want to keep the lowpass activated but then at 4k5
